build.py

```diff
@@ -434,6 +434,12 @@ def walk( project, configs, buildDir ) :
 	help = "The number of build jobs to run in parallel. Defaults to cpu_count."
 )
 
+parser.add_argument(
+	"--compilerRoot",
+	default = "",
+	help = "The root path of the compiler. GCC 6.3.1 is located in \"/opt/rh/devtoolset-6/root\" for CentOS 7."
+)
```

Comment:
I don't think having this as a build argument is helpful. The current expectation is that the compiler is found automatically on `PATH`, but this argument allows you to specify something completely non-matching to that. To be reliable, the argument needs to be truly in control and determine the compiler user to build _every_ project. Also, I wonder what happens with the current default when we end up specifying `-DGCC_INSTALL_PREFIX=""` for the LLVM build?

I had originally assumed (but not communicated well) that we would derive the `{compilerRoot}` variable from a search for `GCC/Clang` on `PATH`. Is there any reason that wouldn't work?

Reply:
Ah yes you were clear here. I'll have a study on how to do this, and I believe the default for `-DGCC_INSTALL_PREFIX=` would be `/usr` for the system, which is in the `config.py` still, but I didn't check if that build successfully on 6.3.1.

Reply:
Alright I am doing a test by running `which c++` and getting the path (including `/usr`) from it. Perhaps this might do the trick, unless you want something more elegant in case of an error?